crates/router/src/core/errors.rs

```diff
@@ -54,8 +54,8 @@ pub enum StorageError {
     DatabaseError(error_stack::Report<storage_errors::DatabaseError>),
     #[error("ValueNotFound: {0}")]
     ValueNotFound(String),
-    #[error("DuplicateValue: {0}")]
-    DuplicateValue(String),
+    #[error("DuplicateValue: {entity} already exists for key {key}")]
+    DuplicateValue { entity: &'static str, key: String },
```

Comment:
Should we be exposing the `key` that we internally use to store the entity? I personally feel leaking an implementation detail is not a good idea.

(I'm specifically speaking with respect to ephemeral key.)

Reply:
Yes exactly my thoughts since we store api-key as a key for a fetching it later in ephemeral key

Reply:
The database errors are not propogated to user, it will be shown only in the logs, the duplicate error in case of ephemeral key is raised only in case of creating the ephemeral key